composer/utils/dist.py

```diff
@@ -235,23 +234,18 @@ def initialize_dist(backend: str, timeout: datetime.timedelta):
         return
 
     if dist.is_initialized():
-        if not dist.get_backend() == backend.lower():
+        if dist.get_backend() != backend.lower() and get_world_size() > 1:
             warnings.warn(f"The requested backend ({backend}) differs from the backend "
```

Comment:
Can you convert this back to a RuntimeError? Another PR made it a warning :disappointed: